fast64_internal/oot/oot_actor.py

```diff
@@ -1,47 +1,410 @@
+	bl_idname = 'object.oot_upgrade_actors'
+	bl_label = "Upgrade Data Now!"
+	bl_options = {'REGISTER', 'UNDO', 'PRESET'}
+
+	version: bpy.props.IntProperty(name="OOT_ObjectProperties Version", default=0)
+	cur_version = 1 # version after property migration
+
+	def execute(self, context):
+		for obj in bpy.context.scene.objects:
+			if obj.fast64.oot.version != self.cur_version:
+				OOTActorProperties.upgrade_object(obj)
+			obj.fast64.oot.version = self.cur_version
+		return {'FINISHED'}
+
+class OOT_SearchChestContentEnumOperator(bpy.types.Operator):
+	bl_idname = "object.oot_search_chest_content_enum_operator"
+	bl_label = "Select Chest Content"
+	bl_property = "itemChest"
+	bl_options = {'REGISTER', 'UNDO'}
+
+	itemChest : bpy.props.EnumProperty(items = ootChestContent, default = '0x48')
 	objName : bpy.props.StringProperty()
 
 	def execute(self, context):
-		obj = bpy.data.objects[self.objName]
-		if self.actorUser == "Transition Actor":
-			obj.ootTransitionActorProperty.actor.actorID = self.actorID
-		elif self.actorUser == "Actor":
-			obj.ootActorProperty.actorID = self.actorID
-		elif self.actorUser == "Entrance":
-			obj.ootEntranceProperty.actor.actorID = self.actorID
-		else:
-			raise PluginError("Invalid actor user for search: " + str(self.actorUser))
+		bpy.data.objects[self.objName].fast64.oot.actor.itemChest = self.itemChest
+		bpy.context.region.tag_redraw()
+		self.report({'INFO'}, "Selected: " + self.itemChest)
+		return {'FINISHED'}
 
+	def invoke(self, context, event):
+		context.window_manager.invoke_search_popup(self)
+		return {'RUNNING_MODAL'}
+
+class OOT_SearchNaviMsgIDEnumOperator(bpy.types.Operator):
+	bl_idname = "object.oot_search_navi_msg_id_enum_operator"
+	bl_label = "Select Message ID"
+	bl_property = "naviMsgID"
+	bl_options = {'REGISTER', 'UNDO'}
+
+	naviMsgID : bpy.props.EnumProperty(items = ootNaviMsgID, default = "0x00")
+	objName : bpy.props.StringProperty()
+
+	def execute(self, context):
+		bpy.data.objects[self.objName].fast64.oot.actor.naviMsgID = self.naviMsgID
 		bpy.context.region.tag_redraw()
-		self.report({'INFO'}, "Selected: " + self.actorID)
+		self.report({'INFO'}, "Selected: " + self.naviMsgID)
 		return {'FINISHED'}
 
 	def invoke(self, context, event):
 		context.window_manager.invoke_search_popup(self)
 		return {'RUNNING_MODAL'}
 
+class OOTActorProperties(bpy.types.PropertyGroup):
+	@staticmethod
+	def upgrade_object(obj):
+		if obj.data is None:
+			if obj.ootEmptyType == 'Scene':
+				print(f"Processing Scene '{obj.name}'...")
+				for roomObj in obj.children:
+					if roomObj.data is None:
+						if roomObj.ootEmptyType == 'Room':
+							# Actor is parented to a room
+							for obj in roomObj.children:
+								upgradeActorInit(obj)
+						elif (roomObj.ootEmptyType == 'Actor' or \
+							roomObj.ootEmptyType == 'Transition Actor' or roomObj.ootEmptyType == 'Entrance'):
+							# Actor is parented to a scene
+							upgradeActorInit(roomObj)
```

Comment:
calling `roomObj` something that may be a room or something else is confusing, I can suggest the uninspired `sceneChildObj` but idk

alternatively you could turn `upgrade_object` into a recursive function which may approximately look like
```py
if obj.data is None:
	if obj.ootEmptyType == 'Scene' (or Room):
		for child in obj.children:
			upgrade_object(child) # recursion
	elif obj is actor stuff:
		upgradeActorInit(obj)
```